lightwood/encoders/time_series/rnn.py

```diff
@@ -61,34 +59,32 @@ def prepare_encoder(self, priming_data, feedback_hoop_function = None):
                         next_tensor, encoder_hidden = self._encoder.forward(data_tensor[tensor_i] , encoder_hidden)
                     else:
                         next_tensor, encoder_hidden = self._encoder.forward(next_tensor.detach(), encoder_hidden)
-                    loss += criterium(next_tensor, data_tensor[tensor_i+1])
+                    loss += criterion(next_tensor, data_tensor[tensor_i+1])
 
 
                 loss = loss
                 average_loss += int(loss)
                 loss.backward()
                 optimizer.step()
-            average_loss = average_loss/total
+
+            average_loss = average_loss/len(priming_data)
+
             if average_loss < self._stop_on_error:
                 break
             if feedback_hoop_function is not None:
                 feedback_hoop_function("epoch [{epoch_n}/{total}] average_loss = {average_loss}".format(average_loss=average_loss, epoch_n=i+1, total=self._train_iters))
 
         self._prepared = True
 
-    def encode_one(self, data, initial_hidden = None, as_list = False, return_next_value = False):
+    def _encode_one(self, data, initial_hidden = None, return_next_value = False):
```

Comment:
why a hidden method?

Reply:
because we only use it internally (in the encoder) and I'd rather keep the encoder's interface standard (e.g. only expose `to`, `prepare_encoder`, `encoder`, `decode` and the constructor).

Is there a situation where you see us using `encode_one` outside of the RNN ts encoder ?

Reply:
ok, I agree